### ISSUE TITLE
Issue: News

### ISSUE THREAD

**Issue by LaocheXe** [e107inc/e107#3034]
I've notice an issue with my site - https://501stlegion-a3.com/index.php  - none of the Tab Names seem to be working, was told to put define('PAGE_NAME', LAN_RSYS_PAGE_RANKS); <--- it used to work, but no longer does.

Now I notice the News section of the site seems to be miss behaving.

![capture1](https://user-images.githubusercontent.com/5818022/36053689-3de0807a-0db8-11e8-895a-04a4c202cd91.JPG)

![capture2](https://user-images.githubusercontent.com/5818022/36053694-438fe7e0-0db8-11e8-825c-ebb2a0827ba6.JPG)

As you can see, in the first picture, in admin area we have a few images set to center, and text - in the news page of that posted news, none of it is being center.

Next, is the News Carousel:
![capture3](https://user-images.githubusercontent.com/5818022/36053720-6c2882fc-0db8-11e8-8bda-249575b22a07.JPG)

It should be loading 4, right now it's loading all....

![capture4](https://user-images.githubusercontent.com/5818022/36053815-f1c75b86-0db8-11e8-8d94-cd42f1de4f30.JPG)

Only thing I have done was update via github - haven't missed with the theme, or anything else.

**Comment by tgtje** [comment]
Can you debug in admin ? look for the path to the language files... (locale folder or root url ? )
ps include  _once(): Failed opening 'e107_plugins/roster/e_shortcode.php' for inclusion

**Comment by LaocheXe** [comment]
@tgtje is that on the front page? I notice it as well - weird thing is, I don't see why it's being loaded on the front page at all, no menus or anything like that.

**Comment by tgtje** [comment]
Hmmm first the language related (found it (i guess) but _needs other peoples reaction too_ if it works..
added   >  ` e107::lan('news',true);`  to line 15 of plug/news/e_menu php

The related news posting (center; left/right) can not replicate..

Just a note : when running e107 dev (debug) front page... > message: Featurebox returned nothing.

**Comment by LaocheXe** [comment]
Yeah not using the featurebox

**Comment by CaMer0n** [comment]
@Deltik This problem appears to be caused by recent changes to `e_tree_model::load() `. 
The 'db_limit' param (and possibly others) is not being respected. 
Rolling back model_class.php corrected the problem @LaocheXe was having with too many items appearing in the carousel.  

Code Sequence: 

    news_carousel_menu.php > e_news_tree::loadJoinActive > self::loadJoin() > e_tree_model::load()

**Comment by Deltik** [comment]
@CaMer0n: How is this possible?  I haven't been able to enable the carousel to reproduce the issue.

If `$this->getParam('db_query') && $class_name && class_exists($class_name)` is true, then the custom tree loading code is executed, including `e_tree_model::prepareSimulatedPagination()`, which parses all possible formats of SQL `LIMIT` and expands the generated `db_limit` param into `db_limit_count` and/or `db_limit_offset` as needed for the PHP-based tree view.

I am also confused by why a simple list uses `e_tree_model`.

**Comment by CaMer0n** [comment]
@Deltik Here's the value of  `$this->getParam('db_query')` at line **3338** of **model_class.php**
```
SELECT  SQL_CALC_FOUND_ROWS n.*, u.user_id, u.user_name, u.user_customtitle, u.user_image, nc.category_id, nc.category_name, nc.category_sef, nc.category_icon FROM #news AS n
			LEFT JOIN #user AS u ON n.news_author = u.user_id
			LEFT JOIN #news_category AS nc ON n.news_category = nc.category_id
			WHERE  1  AND n.news_thumbnail != ''  AND n.news_start < 1518723113 AND (n.news_end=0 || n.news_end>1518723113)
			AND n.news_class REGEXP '(^|,)(12|3|7|253|254|250|251|0)(,|$)' AND NOT (n.news_class REGEXP '(^|,)(255)(,|$)')
		
			ORDER BY n.news_sticky DESC, n.news_datestamp DESC LIMIT 4
```

and here it is again at line 3347 after
**$this->prepareSimulatedPagination();** 
and 
**$this->prepareSimulatedCustomOrdering();**

```
SELECT  SQL_CALC_FOUND_ROWS n.*, u.user_id, u.user_name, u.user_customtitle, u.user_image, nc.category_id, nc.category_name, nc.category_sef, nc.category_icon FROM #news AS n
			LEFT JOIN #user AS u ON n.news_author = u.user_id
			LEFT JOIN #news_category AS nc ON n.news_category = nc.category_id
			WHERE  1  AND n.news_thumbnail != ''  AND n.news_start < 1518723113 AND (n.news_end=0 || n.news_end>1518723113)
			AND n.news_class REGEXP '(^|,)(12|3|7|253|254|250|251|0)(,|$)' AND NOT (n.news_class REGEXP '(^|,)(255)(,|$)')
		
			ORDER BY n.news_sticky DESC, n.news_datestamp DESC 
```

**Comment by Deltik** [comment]
@CaMer0n: Issue was the wrong kind of check used in the count-only condition.  Fixed in 0bbaaf4c6c01bc5a1642402f63fa7e046e614044.  Pull request is #3041.

**Comment by CaMer0n** [comment]
Thank you @Deltik !! 🥇
